### ISSUE TITLE
Why I cannot just add `upgrade-v0.9` branch without dev it?

### ISSUE THREAD

**Issue by sairus7** [JuliaImGui/ImPlot.jl#25]
```julia
import Pkg
Pkg.activate(; temp=true)
Pkg.add(url="https://github.com/wsphillips/ImPlot.jl", rev="upgrade-v0.9")
```
outputs:
```julia
ERROR: LoadError: GitError(Code:ERROR, Class:Submodule, cannot get submodules without a working tree)
```

**Comment by wsphillips** [comment]
hmmm, looks like error due to the submodules I setup to point at cimguipack in the `gen` directory

Does it happen if you clone the repo locally, checkout the branch, and then "add" the directory? I would hope that you don't have to initialize the submodules for normal use since they're only used for development/code gen. It may be package manager not knowing what to do with git submodules when it finds them, which either means I have to set things up differently, or we file an issue upstream

**Comment by sairus7** [comment]
However, I can add `main` branch with no errors.

Seems similar to this: https://github.com/JuliaLang/Pkg.jl/issues/708

**Comment by sairus7** [comment]
> Does it happen if you clone the repo locally, checkout the branch, and then "add" the directory?

I can clone, checkout and `dev` the directory.

**Comment by wsphillips** [comment]
Yes, I don't think the submodule was added to `main`.

From a brief skim, it looks like the work around is to use `git subtree`, manage cimguipack as an artifact, or continue using submodules, but snapshot releases so they can be added from the package manager without issues.

> 
> You can, however, develop the project however you want using submodules, but then at release time, take a snapshot of the entire work tree, including the content of all submodules, and make that the release. I should also point out that your C++ library use case is exactly what artifacts are for:
> 
> https://julialang.org/blog/2019/11/artifacts/

**Comment by sairus7** [comment]
I am using this branch for a quite time with the following install script to make this dependency work from other projects:

```bash
julia --startup-file=no -e 'import Pkg
Pkg.activate(; temp=true)
Pkg.develop(url="https://github.com/wsphillips/ImPlot.jl")'

dir=$pwd
cd ~/.julia/dev/ImPlot
git checkout upgrade-v0.9
cd $dir

julia --project=@. --startup-file=no -e 'import Pkg
Pkg.add(url="https://github.com/Gnimuc/CImGui.jl", rev="master")
Pkg.develop(url="https://github.com/wsphillips/ImPlot.jl")
Pkg.instantiate()'
```

And I am wondering if we can drop submodule at all, since LibCImGUi already has CImGuiPack_jll in its deps?
https://github.com/JuliaImGui/LibCImGui.jl/blob/master/Project.toml

**Comment by wsphillips** [comment]
just make a PR to remove it. it's not worth keeping if it's causing issues with Pkg

**Comment by sairus7** [comment]
closed by #20
